zerver/views/invite.py

```diff
@@ -83,8 +84,15 @@ def invite_users_backend(
             )
         streams.append(stream)
 
-    if len(streams) and not user_profile.can_subscribe_other_users():
-        raise JsonableError(_("You do not have permission to subscribe other users to streams."))
+    if not user_profile.can_subscribe_other_users():
+        if len(streams):
+            raise JsonableError(
+                _("You do not have permission to subscribe other users to streams.")
+            )
+
+        # We would subscribe the invited user to default streams even when the user
+        # inviting them does not have permission to subscribe others.
+        streams = get_default_streams_for_realm(user_profile.realm_id)
```

Comment:
I added this as discussed in [this comment](https://github.com/zulip/zulip/pull/25739#issuecomment-1562181618). We do not want to subscribe users to streams when `stream_ids` is passed as empty explicitly. But in case of `not user_profile.can_subscribe_other_users` we do not allow `stream_ids` to be passed as non-empty list and in such case we do not show any UI for selecting streams to users and thus want the invited users to be subscribed to default streams.

Reply:
I guess @mateuszmandera recent work in https://github.com/zulip/zulip/pull/29556 also provides a very clear user case for having it be empty.

Reply:
And as we also add `subscribe_to_default_streams` parameter, we can probably allow users without permission to subscribe others, to set the parameter as false and not subscribe the new user to default streams. Though as of this PR, we do not allow to user to deselect the `subscribe_to_default_streams` parameter if the user is not allowed to subscribe others to streams.